unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3913 @@
+        wl_list_insert_fake.custom_fake = custom_wl_list_insert;
+        wl_list_remove_fake.custom_fake = custom_wl_list_remove;
+        wl_list_empty_fake.custom_fake = custom_wl_list_empty;
+        wl_array_init_fake.custom_fake = custom_wl_array_init;
+        wl_array_release_fake.custom_fake = custom_wl_array_release;
+        wl_array_add_fake.custom_fake = custom_wl_array_add;
+        wl_signal_init(&m_westonCompositor.output_created_signal);
+        wl_signal_init(&m_westonCompositor.output_destroyed_signal);
+        wl_signal_init(&m_westonCompositor.output_resized_signal);
+        wl_signal_init(&m_westonCompositor.destroy_signal);
+    }
+
+    struct wl_resource *mp_wlResourceDefault = (struct wl_resource *)0xFFFFFF00;
+    struct weston_buffer *mp_westonBuffer = (struct weston_buffer *)0xFFFFFF00;
+
+    struct wl_listener m_listenSurface[OBJECT_NUMBER] = {};
+    struct ivisurface *mp_iviSurface[OBJECT_NUMBER] = {nullptr};
+    struct ivi_layout_surface m_layoutSurface[OBJECT_NUMBER] = {};
+    struct ivi_layout_surface_properties m_layoutSurfaceProperties[OBJECT_NUMBER] = {};
+    struct notification *mp_surfaceNotification[OBJECT_NUMBER] = {nullptr};
+
+    struct ivilayer *mp_iviLayer[OBJECT_NUMBER] = {nullptr};
+    struct ivi_layout_layer m_layoutLayer[OBJECT_NUMBER] = {};
+    struct ivi_layout_layer_properties m_layoutLayerProperties[OBJECT_NUMBER] = {};
+    struct notification *mp_LayerNotification[OBJECT_NUMBER] = {nullptr};
+
+    struct iviscreen *mp_iviScreen[OBJECT_NUMBER] = {nullptr};
+    struct weston_output m_westonOutput[OBJECT_NUMBER] = {};
+
+    struct ivicontroller *mp_iviController[OBJECT_NUMBER] = {nullptr};
+
+    struct ivishell *mp_iviShell = nullptr;
+    struct weston_compositor m_westonCompositor = {};
+
+    struct screen_id_info *mp_screenInfo = nullptr;
+
+    struct wl_client *mp_fakeClient = nullptr;
+
+    static int ms_idAgentModuleValue;
+    static int ms_inputControllerModuleValue;
+    static uint32_t ms_screenIdOffset;
+    static uint32_t ms_screenId;
+    static char *ms_iviClientName;
+    static char *ms_debugScopes;
+    static char *ms_screenName;
+    static int32_t ms_bkgndSurfaceId;
+    static uint32_t ms_bkgndColor;
+    static bool ms_enableCursor;
+    static uint32_t ms_westonConfigNextSection;
+
+    int mp_successResult[1] = {0};
+    int mp_failureResult[1] = {-1};
+};
+
+int ControllerTests::ms_idAgentModuleValue = 0;
+int ControllerTests::ms_inputControllerModuleValue = 0;
+uint32_t ControllerTests::ms_screenIdOffset = 0;
+uint32_t ControllerTests::ms_screenId = 0;
+char *ControllerTests::ms_iviClientName = nullptr;
+char *ControllerTests::ms_debugScopes = nullptr;
+char *ControllerTests::ms_screenName = nullptr;
+int32_t ControllerTests::ms_bkgndSurfaceId = 0;
+uint32_t ControllerTests::ms_bkgndColor = 0;
+bool ControllerTests::ms_enableCursor = false;
+uint32_t ControllerTests::ms_westonConfigNextSection = 0;
```

Comment:
This can be done in ControlerContext class, which would be separated from ControlerTest classe

Reply:
we will consider to do this thing in refactor action in the future